### PR TITLE
chore(LoginPage): a single function to update password and username

### LINKED ISSUE #1462: Login form redundant change handler
Create one change handler to handle all form field updates rather than seperate handlers for username and password.

### PR DESCRIPTION
**Pull request by boaz0**
**What**:

closes #1462 

//cc @ibolton336

### CODE REVIEW

File: packages/patternfly-4/react-core/src/components/LoginPage/LoginForm.test.js [hunk 1]
```diff
@@ -74,3 +74,21 @@ test('LoginForm with rememberMeLabel and rememberMeAriaLabel does not generates
...
+});
+
+test('should call onChangeText callback when is set and password is changed', () => {
+  const mockFn = jest.fn();
+  const view = shallow(
+    <LoginForm onChangePassword={mockFn} rememberMeLabel="Login Form" rememberMeAriaLabel="Login Form" />
+  );
+  view.find('#pf-login-password-id').simulate('change');
+  expect(mockFn).toHaveBeenCalled();
+});
```
Comment (tlabaj) [inline packages/patternfly-4/react-core/src/components/LoginPage/LoginForm.test.js]: should probably add a test case for when the user defines both onChangeUsername/onChangePassword and OnChangeText.

Comment (boaz0) [reply packages/patternfly-4/react-core/src/components/LoginPage/LoginForm.test.js]: You're right. Will do

File: packages/patternfly-4/react-core/src/components/LoginPage/LoginForm.test.js [hunk 2]
```diff
@@ -74,3 +74,42 @@ test('LoginForm with rememberMeLabel and rememberMeAriaLabel does not generates
...
+test('should call onChangeUsername when is set and username is changed', () => {
+  const mockFn = jest.fn();
+  const view = shallow(
+    <LoginForm onChangeUsername={mockFn} rememberMeLabel="Login Form" rememberMeAriaLabel="Login Form" />
+  );
+  view.find('#pf-login-username-id').simulate('change');
+  expect(mockFn).toHaveBeenCalled();
+});
+
+test('should call onChangeUsername callback when is set and password is changed', () => {
```
Comment (tlabaj) [inline packages/patternfly-4/react-core/src/components/LoginPage/LoginForm.test.js]: Do you mean onChangePassword here?

Comment (boaz0) [reply packages/patternfly-4/react-core/src/components/LoginPage/LoginForm.test.js]: Thanks!

File: packages/patternfly-4/react-core/src/components/LoginPage/LoginForm.js [hunk 3]
```diff
@@ -58,18 +60,19 @@ const defaultProps = {
   helperText: null,
   usernameLabel: 'Username',
   usernameValue: '',
-  onChangeUsername: () => undefined,
+  onChangeUsername: null,
   isValidUsername: true,
   passwordLabel: 'Password',
   passwordValue: '',
-  onChangePassword: () => undefined,
+  onChangePassword: null,
```
Comment (dlabaj) [inline packages/patternfly-4/react-core/src/components/LoginPage/LoginForm.js]: Can we keep this as undefined?

Comment (boaz0) [reply packages/patternfly-4/react-core/src/components/LoginPage/LoginForm.js]: Sure we can.

The reason I changed it from `() => undefined` to `null` is that I can't compare functions but `null` and `undefined` I can.

**Comment by jschuler** [comment]
It may seem redundant, but it gives user more flexibility to choose to either define separate event handlers for username and password, or the same. I don't think we should add a third option here, that makes the API more confusing
```
onChangeUsername={this.handleChangeText}
onChangePassword={this.handleChangeText}
```

**Comment by boaz0** [comment]
@jschuler thanks for your feedback.
So you're suggesting to remove the `onChangeText` prop, is that right?

Yeah, I don't mind doing so, I just thought that eventually there is going to be only one prop that handles the events of password and username inputs. 

@tlabaj & @ibolton336  what do you think? Should we discard the `onChangeText` prop and keep the old `onChangePassword` and `onChangeUsername`?

**Comment by jschuler** [comment]
@boaz0 I could see us changing the API to have just one prop, but then we need to inform the user which input field the change came from and this would also be a breaking change

**Comment by dlabaj** [comment]
Talked with @ibolton336 and he's good with the solution purposed by @jschuler Do we just want to add an example instead?

**Comment by boaz0** [comment]
@dlabaj yep, just need to change the example. I will update this PR in the next hours.

**Comment by boaz0** [comment]
@tlabaj no problem - created a new example.

**Comment by jschuler** [comment]
@boaz0 basically it's just one prop changed to point to same callback function, no? I don't see too much value in having a whole other example just for this. If you guys are really convinced we need another example for this then ok... if you agree we don't need please 👍 if you disagree 👎 . I'll defer to the outcome of that :)

**Comment by boaz0** [comment]
@jschuler yes, you're right. It is just one prop change. 

I believe there is trade off:
On one hand, I would like to provide all the possible ways to use any component. On the other hand, it does seem like a redundant to have those two examples.

I wish I had a solid opinion :thinking:  but don't have much of experience to decide :man_shrugging:

Thanks. :smile_cat:

**Comment by jschuler** [comment]
Hi @boaz0 , we have many components with various callback props, I believe in this case it would be too much to showcase this variation to bind 2 callback props to the same function, as otherwise the example is nearly identical to the previous one and this could add noise/confusion to the documentation. Thanks!

**Comment by boaz0** [comment]
No problem. Thank you.

**Pull request closed without merging**